### PR TITLE
Generalize User Object

### PR DESCRIPTION
**Pull request by RohanNagar**
### This PR addresses:

Addresses #95

- Renames `PilotUser` to `User`
- Replaces custom Pilot properties with a `Map<String, Object> properties`. This allows for any number of additional properties to be specified in the JSON, and Thunder will accept it.

### I have verified that:

- [x] All related unit tests have been updated/created
- [x] Integration tests are passing

### Additional Notes

This is a big refactor, please take your time reviewing.

The new user JSON will look like:

```json
{
  "email" : {
    "address" : "test@test.com",
    "verified" : "true",
    "verificationToken" : "hashToken"
  },
  "password" : "12345",
  "properties" : {
    "facebookAccessToken" : "fb",
    "myRandomProperty": 1000,
    "listsShouldWorkToo": ["hello", "world"]
  }
}
```

TODO in the future: Add more tests for different types of additional properties. Will add an issue once this PR is merged.

### CODE REVIEW

File: api/src/main/java/com/sanction/thunder/models/User.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  public int hashCode() {
+    return Objects.hashCode(this.email);
+  }
+
+  @Override
+  public String toString() {
+    return new StringJoiner(", ", "User [", "]")
+        .add(String.format("email=%s", email))
+        .toString();
+  }
```
Comment (NickEckert) [inline api/src/main/java/com/sanction/thunder/models/User.java]: Did you mean to only include the email or should the rest of the properties be here too?

Comment (RohanNagar) [reply api/src/main/java/com/sanction/thunder/models/User.java]: Good catch let me add those

**Comment by RohanNagar** [comment]
@NickEckert updated

File: api/src/test/resources/fixtures/user.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "email" : {
+    "address" : "test@test.com",
+    "verified" : "true",
+    "verificationToken" : "hashToken"
+  },
+  "password" : "12345",
+  "properties" : {
+    "facebookAccessToken" : "fb"
```
[the remark this answers is not in the document]
Comment (RohanNagar) [reply api/src/test/resources/fixtures/user.json]: Good point, I will change this in another iteration. Added #142 to track

**Pull request merged**